src/test/scala/loamstream/googlecloud/CloudSdkDataProcWrapperTest.scala

```diff
@@ -0,0 +1,116 @@
+package loamstream.googlecloud
+
+import org.scalatest.FunSuite
+
+import loamstream.util.BashScript.Implicits.BashPath
+import loamstream.util.ExitCodeException
+import loamstream.util.ExitCodes
+import loamstream.util.Paths
+
+/**
+ * @author clint
+ * Nov 29, 2016
+ */
+final class CloudSdkDataProcWrapperTest extends FunSuite {
+
+  import loamstream.TestHelpers.path
+
+  private val config = GoogleCloudConfig(
+    gcloudBinary = path("/foo/bar/baz"),
+    gsutilBinary = path("/blah/blah/blah"),
+    projectId = "pid",
+    clusterId = "cid",
+    credentialsFile = path("N/A"),
+    numWorkers = 42,
+    //non-default values
+    zone = "z",
+    masterMachineType = "mmt",
+    masterBootDiskSize = 99,
+    workerMachineType = "wmt",
+    workerBootDiskSize = 17,
+    numPreemptibleWorkers = 123,
+    preemptibleWorkerBootDiskSize = 11,
+    properties = "p,r,o,p,s",
+    maxClusterIdleTime = "42m")
+    
+  private val examplePath = Paths.newAbsolute("foo", "bar", "baz")
+
+  private def alwaysFails(tokens: Seq[String]): Int = 42
+
+  private def alwaysSucceeds(tokens: Seq[String]): Int = ExitCodes.success
+
+  test("deleteCluster") {
+    val client = new CloudSdkDataProcWrapper(config, alwaysSucceeds)
+
+    //Shouldn't throw
+    client.stopCluster()
+  }
+
+  test("deleteCluster - should throw") {
+    val client = new CloudSdkDataProcWrapper(config, alwaysFails)
+
+    intercept[ExitCodeException] {
+      client.stopCluster()
+    }
+  }
+
+  test("isClusterRunning") {
+    val client = new CloudSdkDataProcWrapper(config, alwaysSucceeds)
+
+    assert(client.isClusterRunning)
+  }
+
+  test("isClusterRunning - not running") {
+    val client = new CloudSdkDataProcWrapper(config, alwaysFails)
+
+    assert(client.isClusterRunning === false)
+  }
+
+  test("gcloudTokens") {
+    import CloudSdkDataProcWrapper.gcloudTokens
+
+    val tokens = gcloudTokens(config)("foo")("--bar", "Baz")
+
+    val expectedTokens = {
+      Seq(examplePath.render, "beta", "dataproc", "clusters", "foo", "--project", config.projectId, "--bar", "Baz")
+    }
+    
+    assert(tokens === expectedTokens)
```

Comment:
Since command line arguments are (for the most part) rarely order dependent, I wonder if this might be better done as a `Set` and just confirming that all the tokens are present instead of being a perfect-order match? Not sure off the top what the cons would be, but there are probably a couple I'm missing.

Reply:
I hear you, but there are some wrinkles here.  _Some_ of the tokens get their meaning from their positions, like `examplePath.render, "beta", "dataproc", "clusters", "foo"`, while for flags like `--project`, order doesn't matter.

It would be possible to take the first N tokens and expect them to all be in some order, and then treat the rest as an unordered set of pairs.  That would make for a more flexible test, which is better in general, but also be a more complex test, which is worse, all other things being equal.  In this case, I went with the simpler test.